packages/bus-iterator-transform/lib/ActorIteratorTransform.ts

```diff
@@ -0,0 +1,170 @@
+import type { IAction, IActorArgs, IActorOutput, IActorTest, Mediate } from '@comunica/core';
+import { Actor } from '@comunica/core';
+import type { LogicalJoinType, IActionContext, MetadataBindings, MetadataQuads } from '@comunica/types';
+import type * as RDF from '@rdfjs/types';
+import type { AsyncIterator } from 'asynciterator';
+import type { types } from 'sparqlalgebrajs/lib/algebra';
+
+/**
+ * A comunica actor for transform-iterator events.
+ *
+ * Actor types:
+ *  Input:  IActionIteratorTransform: Data that denotes what type of stream is being wrapped,
+ *   what actor produced this stream, and the stream itself
+ * * Test:   <none>
+ * * Output: IActorIteratorTransformOutput: The wrapped stream with the
+ *
+ * @see IActionIteratorTransform
+ * @see IActorIteratorTransformOutput
+ */
+export abstract class ActorIteratorTransform
+  extends Actor<ActionIteratorTransform, IActorTest, ActorIteratorTransformOutput> {
+  public wraps: possibleOperationTypes[];
+  /**
+   * @param args - @defaultNested {<default_bus> a <cc:components/Bus.jsonld#Bus>} bus
+   */
+  public constructor(args: IActorIteratorTransformArgs) {
+    super(args);
+  }
+
+  public async run(action: ActionIteratorTransform):
+  Promise<ActorIteratorTransformOutput> {
+    if (action.type === 'bindings') {
+      const { stream, metadata } = await this.transformIteratorBindings(action);
+      return {
+        type: action.type,
+        operation: action.operation,
+        stream,
+        metadata,
+        originalAction: action.originalAction,
+        context: action.context,
+      };
+    }
+
+    const { stream, metadata } = await this.transformIteratorQuad(action);
+    return {
+      type: action.type,
+      operation: action.operation,
+      stream,
+      metadata,
+      originalAction: action.originalAction,
+      context: action.context,
+    };
+  }
+
+  public async test(
+    action: ActionIteratorTransform,
+  ): Promise<IActorTest> {
+    if (!(this.wraps === undefined) && !this.wraps.includes(action.operation)) {
```

Comment:
I think something like this will also work:
```suggestion
    if (!this.wraps?.includes(action.operation)) {
```

Reply:
It doesn't seem so, it returns true when `this.wraps` is undefined, while that should return false, as `this.wraps === undefined` should make the actor wrap all operations.

Reply:
The explicit `undefined` check can surely go, or be replaced by a falsy check.